spec/src/main/asciidoc/batch_programming_model.adoc

```diff
@@ -1292,10 +1287,46 @@ The following example should make this more clear:
  // WONT WORK! - There is no property 'x' in scope for this injection
  @Inject @BatchProperty(name="x");
 
- // WILL WORK – Gets value 'xVal'
+ // WILL WORK - Gets value 'xVal'
  @Inject @BatchProperty(name="y");
 ----
 
+==== Unmatched property or empty property value does not necessarily get Java default value 
```

Comment:
This is being discussed in Jakarta Config. There is no decision about this yet, but it should soon. It may be interesting to align the behaviours (which value to get in case a config key is not found or empty).

Reply:
Thanks for pointing that out.   Maybe this is overly verbose, but this still strikes me as surprising so at least we have it covered.